app/controllers/master_files_controller.rb

```diff
@@ -282,6 +282,12 @@ def hls_manifest
 
   def caption_manifest
     return head :unauthorized if cannot?(:read, @master_file)
+    caption_id = params[:c_id]
+    @caption_url = if caption_id == 'master_file_caption'
+                     captions_master_file_path
```

Comment:
I don't see a test for this case (id = "master_file_caption").  Could you please add one?